CityOfDetroit/bloom
# refactor: remove components from local uic if matching base uic

# Pull Request Template

## Issue Overview

This PR addresses #3283


## Description

This PR renames Detroit's local UIC to detroit-ui-components. For components where the Detroit version differs, all pages have been changed to pull in a local import for that component from this newly named folder. For components where the Detroit version was the same (or almost the same / the UIC version had improvements Detroit did not), the imports remain the same `@bloom-housing/ui-components` but now it is technically pulling in a "different" version from the base UIC.

Removed local Detroit component, now pulling in the UIC version
* ButtonGroup
* ExpandableText
* LocalizedLink
* Video
* Card
* DashBlock
* HousingCounselor
* InfoCard
* MediaCard
* StatusBar
* StatusItem
* ExygyFooter
* SiteFooter
* DOBField
* DateField
* Dropzone
* Form
* HouseholdMemberForm
* PhoneField
* Select
* Textarea
* TimeField
* All helpers/
* HeaderBadge
* PreferencesList
* Breadcrumbs
* TabNav
* Tabs
* AlertNotice
* ApplicationStatus
* ErrorMessage
* SiteAlert
* StatusAside
* StatusMessage
* Overlay
* ApplicationTimeline
* FormForgotPassword
* ContentAccordion
* ListingMap
* ListingsGroup
* UnitTables
* Contact
* ContactAddress
* GetApplication
* ListingUpdated
* NumberedHeader
* OrDivider
* QuantityRowSection
* ReferralApplication
* SubmitApplication
* DownloadLotteryResults
* EventSection
* FormSignIn
* ForSignInAddPhone
* FormSignInErrorBox
* FormSignInMFACode
* ResendConfirmationModal
* All prototypes/
* ContentSection
* FooterSection
* ListSection
* MarkdownSection
* ResponsiveContentList
* ResponsiveWrappers
* CategoryTable
* MinimalTable
* StackedTable
* Message


Remaining discrepancies (most are very small), kept locally
* Button
* ExpandableContent
* LinkButton
* ActionBlock
* FormCard
* ImageCard
* Tooltip
* ViewItem
* Field
* FieldGroup
* AgPagination
* Heading
* HeadingGroup
* Hero
* PageHeader
* SiteHeader
* StepHeader
* Icon
* FooterNav
* LanguageNav
* ProgressNav
* AlertBox
* Drawer
* LoadingOverlay
* Modal
* AdditionalFees
* ListingCard
* ListingDetailHeader
* ListingDetails
* ListingDividerLine
* WhatToExpect
* FormTerms
* GridSection
* InfoCardGrid
* AgTable
* GroupedTable
* StandardTable
* Description
* Tag


## How Can This Be Tested/Reviewed?

There will be some visual changes. Ideally, they are only minimal and are improvements. Some of the components that Detroit is now using from UIC were actually a little different, but the UIC version is better or more consistent, so this PR and dev are not identical.

[Here](https://docs.google.com/document/d/1CixA6lJdnuqQX0TkmgrVPW_DOnGhoe_I7Wo4HYRXENA/edit?usp=sharing) are some known and IMO acceptable or preferred visual changes.

QA could be relatively large. We need to audit both sites and compare to dev to make sure there are no unwanted or breaking visual changes.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
